core/services/mongodb-remote/src/main/java/com/mongodb/stitch/core/services/mongodb/remote/sync/internal/DataSynchronizer.java

```diff
@@ -2930,9 +2943,11 @@ private LocalSyncWriteModelContainer deleteOneFromRemote(
     } finally {
       lock.unlock();
     }
-    eventDispatcher.emitEvent(nsConfig,
+
+    LocalSyncWriteModelContainer container = desyncDocumentsFromRemote(nsConfig, documentId);
+    container.addLocalChangeEvent(
         ChangeEvents.changeEventForLocalDelete(namespace, documentId, false));
-    return desyncDocumentsFromRemote(nsConfig, documentId);
+    return container;
```

Comment:
[note]  this is a drive-by for another location where we were emitting an event prematurely